src/test/searchqueryparsertest.cpp

```diff
@@ -402,84 +400,143 @@ TEST_F(SearchQueryParserTest, NumericFilterEmpty) {
     EXPECT_FALSE(pQuery->match(pTrack));
-    pTrack->trySetBpm(127.12);
+    pTrack->setBitrate(127);
     EXPECT_TRUE(pQuery->match(pTrack));
     EXPECT_STREQ(
-        qPrintable(QString("bpm >= 127.12")),
-        qPrintable(pQuery->toSql()));
+            qPrintable(QString("bitrate >= 127")),
+            qPrintable(pQuery->toSql()));
 
-    pQuery = m_parser.parseQuery("bpm:<127.12", QString());
-    pTrack->trySetBpm(127.12);
+    pQuery = m_parser.parseQuery("bitrate:<127", QString());
+    pTrack->setBitrate(127);
     EXPECT_FALSE(pQuery->match(pTrack));
-    pTrack->trySetBpm(127.11);
+    pTrack->setBitrate(126);
     EXPECT_TRUE(pQuery->match(pTrack));
     EXPECT_STREQ(
-        qPrintable(QString("bpm < 127.12")),
-        qPrintable(pQuery->toSql()));
+            qPrintable(QString("bitrate < 127")),
+            qPrintable(pQuery->toSql()));
 
-    pQuery = m_parser.parseQuery("bpm:<=127.12", QString());
-    pTrack->trySetBpm(127.13);
+    pQuery = m_parser.parseQuery("bitrate:<=127", QString());
+    pTrack->setBitrate(129);
     EXPECT_FALSE(pQuery->match(pTrack));
-    pTrack->trySetBpm(127.12);
+    pTrack->setBitrate(127);
     EXPECT_TRUE(pQuery->match(pTrack));
     EXPECT_STREQ(
-        qPrintable(QString("bpm <= 127.12")),
-        qPrintable(pQuery->toSql()));
+            qPrintable(QString("bitrate <= 127")),
+            qPrintable(pQuery->toSql()));
+
+    pQuery = m_parser.parseQuery("bitrate:=127", QString());
+    pTrack->setBitrate(129);
+    EXPECT_FALSE(pQuery->match(pTrack));
+    pTrack->setBitrate(127);
+    EXPECT_TRUE(pQuery->match(pTrack));
+    EXPECT_STREQ(
+            qPrintable(QString("bitrate = 127")),
+            qPrintable(pQuery->toSql()));
 }
 
 TEST_F(SearchQueryParserTest, NumericRangeFilter) {
     m_parser.setSearchColumns({"artist", "album"});
-    auto pQuery(
-            m_parser.parseQuery("bpm:127.12-129", QString()));
+    auto pQuery(m_parser.parseQuery("bitrate:127-129", QString()));
+
+    TrackPointer pTrack = newTestTrack();
+    pTrack->setBitrate(125);
+    EXPECT_FALSE(pQuery->match(pTrack));
+    pTrack->setBitrate(127);
+    EXPECT_TRUE(pQuery->match(pTrack));
+    pTrack->setBitrate(129);
+    EXPECT_TRUE(pQuery->match(pTrack));
+
+    EXPECT_STREQ(
+            qPrintable(QString("bitrate BETWEEN 127 AND 129")),
+            qPrintable(pQuery->toSql()));
+}
+
+TEST_F(SearchQueryParserTest, BpmFilter) {
+    m_parser.setSearchColumns({"artist", "album"});
 
+    // Test BpmFilter's MatchModes:
+    // HalveDouble
+    auto pQuery(m_parser.parseQuery("bpm:127", QString()));
     TrackPointer pTrack = newTestTrack();
+    EXPECT_FALSE(pQuery->match(pTrack));
+    pTrack->trySetBpm(127.13);
+    EXPECT_TRUE(pQuery->match(pTrack));
+
+    EXPECT_STREQ(
+            qPrintable(QString("(bpm >= 126.95 AND bpm < 128) OR "
+                               "(bpm BETWEEN 63 AND 64) OR "
+                               "(bpm BETWEEN 253 AND 255)")),
```

Comment:
127 * 2 = 254 
128 * 2 = 256
following the rational above we will find *.0 tracks with just `bpm >= 254.95 AND bpm < 255`
The current implementation lead to a not fully overlapping range: 
```
original 
    127 [----[ 128 
x 2 (my expectation) 
    254 [--------[ 256
implementation
253 [--------] 255 
/2
126,5 [----] 127,5
```

Do we have a reason for this to match tracks, that would not be no longer match after editing the BPM /2? 
I have probably not understand your goal.